suite-common/trading/src/utils.ts

```diff
@@ -427,3 +427,57 @@ export const getTokenSelectableNetworks = (
                 TOKEN_SELECT_SELECTABLE_NETWORKS.includes(n.symbol),
         )
         .map(n => n.symbol);
+
+// reserve defined in networksConfig.ts
+// applies to the native token only
+export const getNetworkReserve = (
+    symbol: NetworkSymbol,
+    contractAddress?: string,
+    isEnabled?: boolean,
+) => {
+    if ((!!contractAddress && contractAddress !== CONTRACT_ADDRESS_FOR_NATIVE_TOKEN) || !isEnabled)
+        return undefined;
+    const network = getNetwork(symbol);
+
+    return network.nativeTokenReserve;
+};
+
+export const getCryptoAmountWithReserve = (
+    symbol: NetworkSymbol,
+    contractAddress: string | undefined,
+    balance: string,
+    amount: string,
+    fee = '0',
+    isNetworkReserveEnabled?: boolean,
+) => {
+    const reserve = getNetworkReserve(symbol, contractAddress, isNetworkReserveEnabled);
+    if (!reserve) return amount;
+
+    if (new BigNumber(balance).minus(amount).gt(new BigNumber(reserve).plus(fee))) {
```

Comment:
nit: I think something like this would be more readable and will create less BigNumber objects.

```suggestion
    const balanceBN = new BigNumber(balance);
    const reservePlusFee = new BigNumber(reserve).plus(fee);
    
    if (balanceBN.minus(amount).gt(reservePlusFee)) {
        return amount;
    }

    const maxAmount = balanceBN.minus(reservePlusFee);
```

Reply:
done